api/proto/teleport/legacy/types/types.proto

```diff
@@ -873,6 +873,23 @@ message AppSpecV3 {
   repeated string UserGroups = 8;
 }
 
+// AppServerOrSAMLIdPServiceProviderV1 holds either an AppServerV3 or a SAMLIdPServiceProviderV1 resource (never both).
+// This is the resource used for WebUI requests to list Applications since we want to list both AppServers and
+// SAMLIdPServiceProviders in the UI.
+message AppServerOrSAMLIdPServiceProviderV1 {
```

Comment:
My thinking here was that since this represents versioned resources (`AppServerV3` & `SAMLIdPServiceProviderV1`), this should be versioned too so that if we then have a `AppServerV4`, we could have a `AppServerOrSAMLIdPServiceProviderV2` which represents it.